app.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+import aws_cdk as cdk
+
+from bde_fdw_rds.bde_fdw_rds_stack import BdeFdwRdsStack
+
+app = cdk.App()
+
+# Parse user provided context for environment parameter (i.e. prod / non-prod)
+environment = app.node.try_get_context("environment")
```

Comment:
[`try_get_context` returns `None` if the value doesn't exist](https://docs.aws.amazon.com/cdk/v2/guide/get_context_var.html), which I guess would be a problem. This is a [long-standing issue](https://github.com/aws/aws-cdk/issues/1771). Fortunately `available_environments.get(environment)` below should throw in that case, but we might want to add a test like `if not environment: raise EnvironmentNotFoundError`.

Reply:
Since only 2 values are allowed, it might be better to do something like this?

```
if environment not in ("prod", "non-prod"):
    raise ValueError("Missing deployment environment. Available options: prod / non-prod")
```

Reply:
Good idea. Updated.